onnx_chainer/functions/loss.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+from onnx import helper
+from onnx.numpy_helper import from_array
+
+from onnx_chainer.onnx_helper import gensym
+from onnx_chainer import onnx_helper
+
+
+def convert_SoftmaxCrossEntropy(
+        func, opset_version, input_names,
+        num_outputs, parameters):
+    # obtain input variable
+    x_var, t_var = func.get_retained_inputs()
+    if len(x_var.shape) != 2:
+        raise NotImplementedError(
+            'onnx-chainer currently handles SoftmaxCrossEntropy only when '
+            'the dimension of input variable x is exactly two.')
+    if np.any(t_var.array == func.ignore_label):
+        raise NotImplementedError(
+            'onnx-chainer currently handles SoftmaxCrossEntropy only when '
+            'ignore_label is not used in input variable t.')
+    if (not func.normalize) or (func.class_weight is not None) or\
+       (func.ignore_label != -1) or (func.reduce != 'mean'):
+        raise NotImplementedError(
+            'onnx-chainer currently handles SoftmaxCrossEntropy only when '
+            'argument parameters are default setting.')
+
+    # create intermediate values
+    nodes = []
+    x, t = input_names
+    y_log = gensym()
+    th = gensym()
+    s0 = gensym()
+    sn = gensym()
+    sr = gensym()
+    depth = gensym()
+    zeroone = gensym()
+
+    nodes.append(helper.make_node(
```

Comment:
Is there any special reason not to use `onnx_helper`? ref https://github.com/chainer/onnx-chainer/blob/master/onnx_chainer/functions/array.py#L53

I thinks the below code can work, not to call `gensym` function, but I'd like to know if you have reasons.
```
nodes.append(onnx_helper.make_node('LogSoftmax', [x], 1))
y_log = nodes[-1].output[0]
```

Reply:
In this case, there are lots of intermediate values. Probably this is a personal preference, but I liked current style because this makes the inputs and outputs of a node more understandable. But if `gensym` is not supposed to be called from outside the helper module, then I would change the this part.

Reply:
I agree `nodes[-1]` is not easy to read. One idea would be defining a helper function and use it like

```
def add_node(*args, **kwargs):
  node = onnx_helper.make_node(*args, **kwargs)
  nodes.append(node)
  return node.output[0]

y_log = add_node(LogSoftmax, [x], 1)
```

However, I think it's better to more this kind of helper function to onnx_helper. Actually, I was planning to do this kind of refactoring by myself. So, how about keeping the code as is in this PR. We will see if I can make it better :)

Reply:
I agree that it would be more helpful if some better function is implemented in the helper module. It sounds good to keep the code as is seems and try to improve later.

Reply:
helper function looks nice, so approve current implementation, thx